src/goto-symex/goto_symex.cpp

```diff
@@ -665,3 +669,57 @@ bool goto_symext::constant_propagate_delete_char_at(
 
   return true;
 }
+
+bool goto_symext::constant_propagate_trim(
+  statet &state,
+  symex_assignt &symex_assign,
+  const function_application_exprt &f_l1)
+{
+  const auto &f_type = to_mathematical_function_type(f_l1.function().type());
+  const auto &length_type = f_type.domain().at(0);
+  const auto &char_type = to_pointer_type(f_type.domain().at(1)).subtype();
+
+  const refined_string_exprt &s = to_string_expr(f_l1.arguments().at(2));
+  const auto s_data_opt = try_evaluate_constant_string(state, s.content());
+
+  if(!s_data_opt)
+    return false;
+
+  const array_exprt &string_data = s_data_opt->get();
+  auto &operands = string_data.operands();
+
+  auto is_not_whitespace = [](const exprt &expr) {
+    auto &constant_value = to_constant_expr(expr);
+    char character = numeric_cast_v<int>(constant_value);
+    return character > ' ';
+  };
+
+  // Note where the trim would start/end.
+  auto end_iter =
+    std::find_if(operands.rbegin(), operands.rend(), is_not_whitespace);
+  auto start_iter =
+    std::find_if(operands.begin(), operands.end(), is_not_whitespace);
+
+  // If we haven't found anything, just want to copy the whole array.
+  if(start_iter == operands.end())
+    start_iter = operands.begin();
```

Comment:
:no_entry_sign: this seems wrong, if there is no character which is not a whitespace, then the result should be empty

Reply:
This actually is producing an empty string, but it's not obvious (`end_iter` must have traversed the whole string too, and so equals `operands.begin()`). Suggest instead do

```
exprt::operandst new_operands;
if(start_iter != operands.end())
  new_operands = exprt::operandst{start_iter, end_iter.base()};
```